test/nginx_aaad_proxy_test.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+from __future__ import print_function
+import socket
+import json
+import sys
+import os
+import requests
+import base64
+import time
+
+
+def get_data(url, headers):
+    result = requests.get(url, headers=headers, auth=('internal_test_user', 'internal_test_user'))
+    return result.status_code
+
+
+def put_data(url, headers, payload):
+    result = requests.put(url, data=json.dumps(payload), headers=headers)
+    return (result.status_code)
+
+
+def delete_data(url, headers, payload):
+    result = requests.delete(url, data=json.dumps(payload), headers=headers, auth=('internal_test_user', 'internal_test_user'))
+    return (result.status_code)
+
+
+def test_proxy_common_user(environment):
+
+    headers = {"Content-Type": "application/json"}
+    # Test GET Access for common_access
+    app_url = app_url = 'http://' + environment + ':4080/marathon/ui/'
+    result_code = get_data(app_url, headers)
+    assert(result_code == 200)
+    print("GET Passed for marathon/ui/")
+
+    app_url = app_url = 'http://' + environment + ':4080/marathon/public/api/'
+    result_code = get_data(app_url, headers)
+    assert(result_code == 200)
+    print("GET Passed for marathon/public/api")
+
+    app_url = app_url = 'http://' + environment + ':4080/marathon/v2/tasks/'
+    result_code = get_data(app_url, headers)
+    assert(result_code == 200)
+    print("GET Passed for marathon/v2/tasks/\n")
+
+
+def test_proxy_user_valid_permissions(environment):
```

Comment:
This method seems monolithic and testing too many stuff. If it fails debugging becomes tougher. Can you break it into different test cases, maybe one for each HTTP method. Also separate our the multiple endpoints ( app_urls) into separate test cases.